src/app/pages/checkout-payment/payment-concardis-creditcard-cvc-detail/payment-concardis-creditcard-cvc-detail.component.ts

```diff
@@ -0,0 +1,153 @@
+          error => {
+            this.scriptLoaded = false;
+            this.errorMessage.general.message = error;
+            this.cd.detectChanges();
+          }
+        );
+    }
+  }
+
+  isCvcExpired() {
+    // if cvc last updated timestamp is less than 25 minutes then return false
+    if (this.paymentInstrument.parameters) {
+      const cvcLastUpdatedAttr =
+        this.paymentInstrument.parameters &&
+        this.paymentInstrument.parameters.find(attribute => attribute.name === 'cvcLastUpdated');
+
+      if (cvcLastUpdatedAttr) {
+        const cvcLastUpdatedValue = cvcLastUpdatedAttr.value ? cvcLastUpdatedAttr.value.toString() : undefined;
+        if (cvcLastUpdatedValue) {
+          const cvcDate = new Date(cvcLastUpdatedValue);
+          const currentDate = new Date();
+          const diffAsMinutes = (currentDate.getTime() - cvcDate.getTime()) / (1000 * 60);
+          if (diffAsMinutes <= parseInt(this.validityTimeInMinutes, 10)) {
+            return false;
+          }
+        }
+      }
+    }
+    return true;
+  }
+
+  /**
+   * call back function to submit data
+   */
+  submitCallback(error) {
+    if (error) {
+      // map error messages
+      if (typeof error.message !== 'string' && error.message.properties) {
+        this.errorMessage.cvc =
+          error.message.properties && error.message.properties.find(prop => prop.key === 'verification');
+        if (this.errorMessage.cvc && this.errorMessage.cvc.code) {
+          this.errorMessage.cvc.messageKey = this.getErrorMessage(
+            this.errorMessage.cvc.code,
+            'credit_card',
+            'cvc',
+            this.errorMessage.cvc.message
+          );
+        }
+      }
+    } else {
+      // update cvcLastUpdated to current timestamp
+      const param =
+        this.paymentInstrument.parameters &&
+        this.paymentInstrument.parameters.map(attr => ({ name: attr.name, value: attr.value }));
+      if (param.find(attribute => attribute.name === 'cvcLastUpdated')) {
+        param.find(attribute => attribute.name === 'cvcLastUpdated').value = new Date().toISOString();
+      } else {
+        param.push({ name: 'cvcLastUpdated', value: new Date().toISOString() });
+      }
+      // Replacing encoded paymentInstrumentId with Token for put request
+      param.find(attribute => attribute.name === 'paymentInstrumentId').value = this.paymentInstrument.parameters.find(
+        attribute => attribute.name === 'token'
+      ).value;
+      const pi: PaymentInstrument = {
+        id: this.paymentInstrument.id,
+        urn: this.paymentInstrument.urn,
+        accountIdentifier: this.paymentInstrument.accountIdentifier,
+        parameters: param,
+        paymentMethod: this.paymentInstrument.paymentMethod,
+      };
+
+      this.checkoutFacade.updateCvcLastUpdated(pi);
+    }
+  }
+
+  renewCVCDetails() {
+    if (this.paymentInstrument.parameters) {
+      const tokenAttr =
+        this.paymentInstrument.parameters &&
```

Comment:
use the ? operator for optional chaining https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html

Reply:
Done.